### PR TITLE
Clientset release 1.3

### PR DESCRIPTION
**Pull request by caesarxuchao**
This PR creates the release 1.3 client set. We'll keep updating this client set until we cut release 1.3. In the meantime, the release 1.2 client set will be locked.

@lavalamp

### CODE REVIEW

**Comment by lavalamp** [comment]
It seems like some autogenerator needs to be run or something? This isn't nearly big enough :)

**Comment by caesarxuchao** [comment]
The typed clients are kept separately from the client set and they are not regenerated for a new release. I think this is the correct behavior?

**Comment by lavalamp** [comment]
No-- because if we change a typed client, we want the 1.2 version to stay
the same.

**Comment by caesarxuchao** [comment]
Right. I only thought about the API changes and missed the potential changes in the client code.

Currently the typed clients live in `pkg/client/typed/<group>/<version>/`, how about we change it to `pkg/client/typed/<release-version>/<group>/<version>`?

**Comment by lavalamp** [comment]
How about  pkg/client/<release-version>/typed/<group>/<version>?

Or maybe omit 'typed' entirely?

**Comment by caesarxuchao** [comment]
@lavalamp. See #23608. It implements what you suggested.

**Comment by caesarxuchao** [comment]
@k8s-bot unit test this, IGNORE ISSUE

**Comment by caesarxuchao** [comment]
The last commit add a basic e2e test for 1.3. @lavalamp PTAL.

**Comment by caesarxuchao** [comment]
Rebased to pull some e2e test changes.

File: test/e2e/generated_clientset.go [hunk 1]
```diff
@@ -126,31 +154,83 @@ var _ = KubeDescribe("Generated release_1_2 clientset", func() {
...
+		w, err := podClient.Watch(options)
+		if err != nil {
+			Failf("Failed to set up watch: %v", err)
+		}
+
+		By("submitting the pod to kubernetes")
+		// We call defer here in case there is a problem with
+		// the test so we can ensure that we clean up after
+		// ourselves
+		defer podClient.Delete(pod.Name, api.NewDeleteOptions(0))
```
Comment (lavalamp) [inline test/e2e/generated_clientset.go]: I would do the defer after the create, but I guess it doesn't really matter in this case.

File: test/e2e/generated_clientset.go [hunk 2]
```diff
@@ -126,31 +154,83 @@ var _ = KubeDescribe("Generated release_1_2 clientset", func() {
...
+		}
+		Expect(len(pods.Items)).To(Equal(0))
+	})
+})
+
+var _ = KubeDescribe("Generated release_1_3 clientset", func() {
+	framework := NewDefaultFramework("clientset")
+	It("should create pods, delete pods, watch pods", func() {
+		podClient := framework.Clientset_1_3.Core().Pods(framework.Namespace.Name)
+		By("creating the pod")
```
Comment (lavalamp) [inline test/e2e/generated_clientset.go]: I think "constructing" or "assembling" would be less confusing.

File: test/e2e/generated_clientset.go [hunk 3]
```diff
@@ -126,31 +154,83 @@ var _ = KubeDescribe("Generated release_1_2 clientset", func() {
...
+		options = api.ListOptions{
+			LabelSelector:   selector,
+			ResourceVersion: pods.ListMeta.ResourceVersion,
+		}
+		w, err := podClient.Watch(options)
+		if err != nil {
+			Failf("Failed to set up watch: %v", err)
+		}
+
+		By("submitting the pod to kubernetes")
```
Comment (lavalamp) [inline test/e2e/generated_clientset.go]: "creating the pod"

**Comment by lavalamp** [comment]
LGTM, just a couple tiny nits.

**Comment by caesarxuchao** [comment]
Thank you. Comments addressed. I'm applying the LGTM.

**Comment by caesarxuchao** [comment]
Adding a priority label, I have PRs depending on this one.

**Pull request merged**